### PR TITLE
[WIP] Defer package activation

### PR DESCRIPTION
**Pull request by maxbrunsfeld**

### CODE REVIEW

**Comment by Zireael07** [comment]
Do I spy a quicker startup?

**Comment by benogle** [comment]
> Do I spy a quicker startup?

Yep! :boom:

**Comment by maxbrunsfeld** [comment]
> Do I spy a quicker startup?

This one will have a more noticeable impact if you have lots of packages installed.

File: src/package.coffee [hunk 1]
```diff
@@ -461,6 +459,26 @@ class Package
   activationShouldBeDeferred: ->
     @hasActivationCommands() or @hasActivationHooks()
 
+  PackagesToActivateImmediately = [
+    # core
+    'status-bar',
+    'tabs',
+    'tree-view',
+    'wrap-guide',
```
Comment (steelbrain) [inline src/package.coffee]: Can we please have `notifications` here as well? It'll fix https://github.com/atom/atom/issues/9522

**Comment by Globegitter** [comment]
@maxbrunsfeld Is there anything specific holding up this PR? This looks like it would be great to get in :)

**Comment by Stanzilla** [comment]
@as-cii is this PR still needed/coming?

**Comment by as-cii** [comment]
We might still want to explore a variation of it in the future, but we could probably close this for the time being. 

@maxbrunsfeld: please, let me know if you think we should reopen this. 👍

**Pull request closed without merging**